### PR TITLE
Create GKE diagnoser command base and update vendor

### PR DESCRIPTION
**Pull request by songrx1997**
Create gke-diagnoser CLI base
 Add required dependencies for gke-diagnoser  and run go mod vendor

### CODE REVIEW

File: cmd/gke-diagnoser/app/command/root.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "kubectl check-gke-ingress",
+	Short: "kubectl-check-gke-ingress is a kubectl tool to check the correctness of kubernetes resources.",
+	Long:  "kubectl-check-gke-ingress is a kubectl tool to check the correctness of kubernetes resources.",
```
Comment (swetharepakula) [inline cmd/gke-diagnoser/app/command/root.go:29]: nit: kubectl check-gke-ingress

maybe add correctness of ingress and ingress related resources.

File: cmd/gke-diagnoser/app/command/root.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Fprintf(os.Stderr, "Hello, this is kubectl-check-gke-ingress")
```
Comment (swetharepakula) [inline cmd/gke-diagnoser/app/command/root.go:31]: why stderr?

Comment (swetharepakula) [reply cmd/gke-diagnoser/app/command/root.go:31]: possibly can we log something like starting gke-check-ingress?

**Review by swetharepakula — COMMENTED** [review]
some nits, otherwise looks good.

File: cmd/gke-diagnoser/app/command/root.go (hunk 1 above)
Comment (songrx1997) [reply cmd/gke-diagnoser/app/command/root.go]: Done

File: cmd/gke-diagnoser/app/command/root.go (hunk 2 above)
Comment (songrx1997) [reply cmd/gke-diagnoser/app/command/root.go]: Done

File: vendor/modules.txt [hunk 3]
```diff
@@ -91,10 +91,17 @@ github.com/googleapis/gax-go/v2/internal
 # github.com/imdario/mergo v0.3.7
 ## explicit
 github.com/imdario/mergo
+<<<<<<< HEAD
```
Comment (swetharepakula) [inline vendor/modules.txt:94]: fix the merge conflict in this file

File: go.mod [hunk 4]
```diff
@@ -6,8 +6,14 @@ require (
 	github.com/GoogleCloudPlatform/k8s-cloud-provider v1.20.0
 	github.com/google/go-cmp v0.5.9
 	github.com/kr/pretty v0.2.0
+<<<<<<< HEAD
```
Comment (swetharepakula) [inline go.mod:9]: fix the merge conflict in this file

File: go.sum [hunk 5]
```diff
@@ -236,9 +236,17 @@ github.com/ianlancetaylor/demangle v0.0.0-20181102032728-5e5cf60278f6/go.mod h1:
 github.com/ianlancetaylor/demangle v0.0.0-20200824232613-28f6c0f3b639/go.mod h1:aSSvb/t6k1mPoxDqO4vJh6VOCGPwU4O0C2/Eqndh1Sc=
 github.com/imdario/mergo v0.3.7 h1:Y+UAYTZ7gDEuOfhxKWy+dvb5dRQ6rJjFSdX2HZY1/gI=
 github.com/imdario/mergo v0.3.7/go.mod h1:2EnlNZ0deacrJVfApfmtdGgDfMuh/nq6Ok1EcJh5FfA=
+<<<<<<< HEAD
```
Comment (swetharepakula) [inline go.sum:239]: fix merge conflict in the file

File: go.sum [hunk 6]
```diff
@@ -321,6 +329,19 @@ github.com/sirupsen/logrus v1.2.0/go.mod h1:LxeOpSwHxABJmUn/MG1IvRgCAasNZTLOkJPx
 github.com/sirupsen/logrus v1.4.2/go.mod h1:tLMulIdttU9McNUspp0xgXVQah82FyeX6MwdIuYE2rE=
 github.com/sirupsen/logrus v1.6.0/go.mod h1:7uNnSEd1DgxDLC74fIahvMZmmYsHGZGEOFrfsX/uA88=
 github.com/spaolacci/murmur3 v0.0.0-20180118202830-f09979ecbc72/go.mod h1:JwIasOWyU6f++ZhiEuf87xNszmSA2myDM2Kzu9HwQUA=
+<<<<<<< HEAD
```
Comment (swetharepakula) [inline go.sum:332]: fix merge conflict in this file.

**Review by swetharepakula — COMMENTED** [review]
Code looks good. Only change needed is addressing the unresolved merge conflicts in dependency related files (go.mod, go.sum, modules.txt).

**Review by swetharepakula — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**